File: spark/learning-spark-lightning-fast-big-data-analysis/jvm/src/main/scala/redocpot/spark/PassFunction.scala
package scala.redocpot.spark

import org.apache.spark.SparkConf
import org.apache.spark.SparkContext
import org.apache.spark.SparkContext._

object PassFunction {
  def hasStr(s : String)(line : String) : Boolean = {
    line.contains(s)
  }

  def main(args : Array[String]) : Unit = {
    val conf = new SparkConf().setAppName("PassFunction").setMaster("local[*]")
    val sc = new SparkContext(conf)
    val textFile = sc.textFile("/Users/me/Code/spark-learning/learning-spark-lightning-fast-big-data-analysis/pom.xml")
    //val xmlLines = textFile.filter(line => line.contains("xml"))
    val xmlLines = textFile.filter(hasStr("xml"))
    println(xmlLines.first)
    sc.stop
  }
}

File: spark/learning-spark-lightning-fast-big-data-analysis/jvm/src/main/java/redocpot/spark/PassFunction.java
package redocpot.spark;

import org.apache.spark.SparkConf;
import org.apache.spark.api.java.JavaRDD;
import org.apache.spark.api.java.JavaSparkContext;
import org.apache.spark.api.java.function.Function;

public class PassFunction {
    public static void main(String[] args) {
        SparkConf conf = new SparkConf()
            .setMaster("local")
            .setAppName("PassFunction");
        JavaSparkContext sc = new JavaSparkContext(conf);

        JavaRDD<String> textFile = sc.textFile(
                "/Users/me/Code/spark-learning/learning-spark-lightning-fast-big-data-analysis/pom.xml");
        JavaRDD<String> xmlLines = textFile.filter(
                new Function<String, Boolean>() {
                    public Boolean call(String s) {
                        return s.contains("xml");
                    }
                }
                );
        xmlLines.foreach(line -> {
            System.out.println(line);
        });

        sc.stop();
    }
}
